### PR TITLE
rbi/core: fix type of Array#combination

### LINKED ISSUE #1130: Array#combination has wrong type
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0AT.reveal_type(%5B1%2C2%2C3%5D.combination(2)))

```ruby
# typed: true

T.reveal_type([1,2,3].combination(2))
```

#### Observed output

```
editor.rb:3: Revealed type: T::Enumerator[Integer] https://srb.help/7014
     3 |T.reveal_type([1,2,3].combination(2))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  From:
    editor.rb:3:
     3 |T.reveal_type([1,2,3].combination(2))
                      ^^^^^^^^^^^^^^^^^^^^^^
Errors: 1
```

#### Expected behavior

I expect this type to be `T::Enumerator[T::Array[Integer]]` (`to_a` in example below to get pry to print values)
```
[1] pry(main)> [1,2,3].combination(2).to_a
=> [[1, 2], [1, 3], [2, 3]]
```

### PR DESCRIPTION
**Pull request by carl-stripe**
Fixes #1130

### CODE REVIEW

File: rbi/core/array.rbi [hunk 1]
```diff
@@ -150,7 +150,7 @@ class Array < Object
     params(
         arg0: Integer,
     )
-    .returns(T::Enumerator[Elem])
+    .returns(T::Enumerator[T::Array[Elem]])
```
Comment (jez) [inline rbi/core/array.rbi]: @carl-stripe can you update both halves of the overload? or is there some reason why the first overload is still correct?

Comment (jez) [reply rbi/core/array.rbi]: This line: https://github.com/sorbet/sorbet/pull/1131/files#diff-41b3206380d3b7469fde9a3fc6c8129eR147

Comment (jez) [reply rbi/core/array.rbi]: jk i see that that half of the overload is actually correct.

Comment (carl-stripe) [reply rbi/core/array.rbi]:
```
[1] pry(main)> [1,2,3].combination(2){|x| }
=> [1, 2, 3]
```

**Pull request merged**